examples/js/loaders/GLTFLoader.js

```diff
@@ -12,368 +16,1836 @@ THREE.GLTFLoader.prototype = {
 
 	constructor: THREE.GLTFLoader,
 
-	load: function ( url, onLoad, onProgress, onError ) {
+	load: function( url, onLoad, onProgress, onError ) {
 
 		var scope = this;
 
+		var baseUrl = this.baseUrl && ( typeof this.baseUrl === "string" ) ? this.baseUrl : THREE.Loader.prototype.extractUrlBase( url );
+
 		var loader = new THREE.XHRLoader( scope.manager );
-		loader.load( url, function ( text ) {
+		loader.load( url, function( text ) {
 
-			onLoad( scope.parse( JSON.parse( text ) ) );
+			scope.parse( JSON.parse( text ), onLoad, baseUrl );
 
 		}, onProgress, onError );
 
 	},
 
-	setCrossOrigin: function ( value ) {
+	setCrossOrigin: function( value ) {
 
 		this.crossOrigin = value;
 
 	},
 
-	parse: function ( json ) {
+	setBaseUrl: function( value ) {
```

Comment:
I've been renaming this method to `setPath` in other loaders:

https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/OBJLoader.js#L56-L60